playground/backend/pkg/fs_tool/java_fs.go

```diff
@@ -0,0 +1,111 @@
+package fs_tool
+
+import (
+	"fmt"
+	"io/fs"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+const (
+	executableFilesFolderName = "executable_files"
+	srcFolder                 = "src"
+	binFolder                 = "bin"
+)
+
+type JavaFileSystemService struct{}
+
+func (jFS *JavaFileSystemService) PrepareFolders() error {
+	// Create executable_files/src folder
+	err := os.MkdirAll(filepath.Join(executableFilesFolderName, srcFolder), fs.ModePerm)
+	if err != nil {
+		return err
+	}
+
+	// Create executable_files/bin folder
+	err = os.MkdirAll(filepath.Join(executableFilesFolderName, binFolder), fs.ModePerm)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (jFS *JavaFileSystemService) DeletePreparedFolders() error {
+	// Delete executable_files/src folder
+	err := os.Remove(filepath.Join(executableFilesFolderName, srcFolder))
+	if err != nil {
+		return err
+	}
+
+	// Delete executable_files/bin folder
+	err = os.Remove(filepath.Join(executableFilesFolderName, binFolder))
+	if err != nil {
+		return err
+	}
+
+	err = os.Remove(executableFilesFolderName)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (jFS *JavaFileSystemService) CreateExecutableFile(code string, pipelineId int) (string, error) {
+	srcFolderPath := filepath.Join(executableFilesFolderName, srcFolder)
+	if _, err := os.Stat(srcFolderPath); os.IsNotExist(err) {
+		return "", err
+	}
+	fileName := getFileName(pipelineId, srcFolder)
+	filePath := filepath.Join(srcFolderPath, fileName)
+	err := ioutil.WriteFile(filePath, []byte(code), 0600)
+	if err != nil {
+		return "", err
+	}
+	return filePath, nil
+}
+
+func (jFS *JavaFileSystemService) DeleteExecutableFile(pipelineId int) error {
+	return deleteFile(pipelineId, srcFolder)
+}
+
+func (jFS *JavaFileSystemService) DeleteCompiledFile(pipelineId int) error {
+	return deleteFile(pipelineId, binFolder)
+}
+
+// deleteFile delete file by pipelineId
+func deleteFile(pipelineId int, folder string) error {
+	folderPath := getFilePath(folder)
+	if _, err := os.Stat(folderPath); os.IsNotExist(err) {
+		return err
```

Comment:
Should we return an error if dir does not exist? How can we catch this scenario outside this method? I mean, how can we understand what specific error returned?

Reply:
i think it's not error scenario

Reply:
I know in Golang people often use bool for found/not found scenario

Reply:
Add bool variable as a returning variable for delete methods